packages/cli/src/util/google-sheet-parser.ts

```diff
@@ -0,0 +1,153 @@
+ * This implementation splits on newline and commas and then sanitizes each cell
+ * by trimming whitespace and removing extraneous surrounding quotes.
+ *
+ * @param csvText - The CSV text.
+ * @returns A 2D array representing the rows and cells of the CSV.
+ */
+function parseCSV(csvText: string): string[][] {
+  return csvText
+    .split('\n')
+    .filter(Boolean)
+    .map(row => row.split(',').map(cell => sanitizeCell(cell)));
+}
+
+/**
+ * Removes surrounding quotes from a cell value if present.
+ * For example: '"GB"' becomes 'GB'
+ *
+ * @param cell - The cell value to sanitize.
+ * @returns The sanitized cell value.
+ */
+function sanitizeCell(cell: string): string {
+  const trimmed = cell.trim();
+  if (trimmed.startsWith('"') && trimmed.endsWith('"')) {
+    return trimmed.substring(1, trimmed.length - 1);
+  }
+  return trimmed;
+}
+
+/**
+ * Retrieves values from a public Google Sheet.
+ * This function constructs its own URL to fetch the CSV export of a public sheet.
+ *
+ * @param spreadsheetId - The ID of the Google Sheet.
+ * @param range - The range of cells to fetch (ignored in this simple implementation).
+ * @param worksheetName - (Optional) The worksheet name (defaults to "Sheet1").
+ * @returns A promise that resolves to a 2D array of cell values.
+ */
+export async function getPublicSheetValues(
+  spreadsheetId: string,
+  range: string,
+  worksheetName: string = 'Sheet1'
+): Promise<string[][]> {
+  // Construct the CSV export URL for the public Google Sheet.
+  const url = `https://docs.google.com/spreadsheets/d/${spreadsheetId}/gviz/tq?tqx=out:csv&sheet=${encodeURIComponent(
+    worksheetName
+  )}`;
+
+  // Use undici's request to fetch the CSV data.
+  const { statusCode, body } = await request(url);
+  if (statusCode < 200 || statusCode >= 300) {
+    throw new Error(`Failed to fetch public Google Sheet: Status code ${statusCode}`);
+  }
+
+  // Read the response body as text.
+  const csvText = await body.text();
+  const data = parseCSV(csvText);
+
+  // In this simple implementation, the entire CSV is returned.
+  // If you need to support a specific range, you could slice or filter the data here.
+  return data;
+}
+
+/**
+ * Parses a Google Sheet into a mapping.
+ *
+ * @param sheetId - The Google Sheet ID or URL.
+ * @param options - Options for parsing:
+ *   - range: The range to fetch (default: 'A1:Z'). (Ignored: we fetch the entire sheet.)
+ *   - nested: If true, creates nested objects from each row (using all columns except the key column).
+ *             If false, uses a single value from a specified value column.
+ *   - keyColumn: The column to use as key (either index or header name).
+ *   - valueColumn: (Required if nested is false) The column to use as value (either index or header name).
+ *   - worksheetName: Optional worksheet name.
+ *
+ * @returns A promise that resolves to the mapping object.
+ */
+export async function parseGoogleSheet(
```

Comment:
I'd love to see some unit tests around this - even at this early state - because this is the meat of the implementation.

If I was writing this feature, I'd rename the function to `parseValues()`, pass the values directly as an array (ie, do the sheet loading stuff separately and divorce the "download data from sheets" and "process sheet data steps" entirely). Then I'd write a bunch of unit tests against parseValues with different options and datasets, to really make sure (to myself, my reviewer, and to future developers) that the parsing functionality was working properly.